x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/nulls/Coalesce.java

```diff
@@ -149,15 +150,17 @@ public Block.Ref eval(Page page) {
                         1,
                         IntStream.range(0, page.getBlockCount()).mapToObj(b -> page.getBlock(b).filter(positions)).toArray(Block[]::new)
                     );
-                    for (EvalOperator.ExpressionEvaluator eval : evaluators) {
```

Comment:
:+1:

This strategy is *even* worse with this work around of copying to filter. Oh well, we know coalesce and case ain't fast yet.